test/unit/domains.server.test.ts

```diff
@@ -0,0 +1,221 @@
+    const domain = `${subDomain}.${rootDomain}`;
+    const value = '192.168.0.1';
+    const data: Domain = {
+      type: RecordType.A,
+      name: domain,
+      value,
+      username: 'NOT EXIST',
+    };
+
+    await expect(createUserDomain(data)).rejects.toThrow();
+  });
+
+  test('updateUserDomain() updates a record in Route 53 and DB', async () => {
+    const subDomain = randomWord();
+    const domain = `${subDomain}.${rootDomain}`;
+    const value = '192.168.0.1';
+    let data: Domain = {
+      type: RecordType.A,
+      name: domain,
+      value,
+      username,
+    };
+
+    const result = await createUserDomain(data);
+
+    const updatedValue = '2001:db8:3333:4444:5555:6666:7777:8888';
+    data.type = RecordType.AAAA;
+    data.value = updatedValue;
+    data.id = result.id;
+    data.description = description;
+    data.course = course;
+    data.ports = ports;
+
+    const updatedResult = await updateUserDomain(data);
+
+    expect(updatedResult.type).toEqual('AAAA');
+    expect(updatedResult.value).toEqual(updatedValue);
+    expect(updatedResult.description).toEqual(description);
+    expect(updatedResult.course).toEqual(course);
+    expect(updatedResult.ports).toEqual(ports);
+    expect(updatedResult.updatedAt.getTime()).toBeGreaterThan(result.createdAt.getTime());
+    expect(updatedResult.expiresAt.getTime()).toBeGreaterThan(result.updatedAt.getTime());
+    expect(result.status.length).toBeGreaterThan(0);
+  });
+
+  test('updateUserDomain() throws when attempting to update non existing ID in DB', async () => {
+    let data: Domain = {
+      type: RecordType.A,
+      name: `update.${rootDomain}`,
+      value: '192.168.0.2',
+      id: 0,
+      username,
+      description,
+      course,
+      ports,
+    };
+
+    await expect(updateUserDomain(data)).rejects.toThrow();
+  });
+
+  test('deleteUserDomain() deletes a record in Route 53 and DB', async () => {
+    const subDomain = randomWord();
+    const domain = `${subDomain}.${rootDomain}`;
+    const value = '192.168.0.1';
+    let data: Domain = {
+      type: RecordType.A,
+      name: domain,
+      value,
+      username,
+    };
+
+    const result = await createUserDomain(data);
+    data.id = result.id;
+
+    const deleteResult = await deleteUserDomain(data);
+    expect(deleteResult.id).toEqual(result.id);
+    expect(deleteResult.name).toEqual(result.name);
+    expect(deleteResult.type).toEqual(result.type);
+    expect(deleteResult.value).toEqual(result.value);
+    expect(deleteResult.status.length).toBeGreaterThan(0);
+  });
+
+  test('deleteUserDomain() throws when attempting to delete non existing ID in DB', async () => {
+    let data: Domain = {
+      type: RecordType.A,
+      name: `delete.${rootDomain}`,
+      value: '192.168.0.3',
+      id: 0,
+    };
+
+    await expect(deleteUserDomain(data)).rejects.toThrow();
+  });
+
+  test('isDomainExpired() deletes an expired record in Route 53 and DB ', async () => {
```

Comment:
This is confusing.  What is `isDomainExpired()`?

Reply:
Will be updated. It should be `removeIfExpired`